src/main/java/adeuxpas/back/datainit/UserDatabaseSeeder.java

```diff
@@ -0,0 +1,264 @@
+import adeuxpas.back.repository.PreferredScheduleRepository;
+import adeuxpas.back.repository.UserRepository;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Component;
+
+import java.time.*;
+import java.util.*;
+
+/**
+ * Class responsible for seeding the database with initial sample data.
+ * It contains methods to seed different entities in the database, such as users.
+ * <p>
+ * The {@code seedDatabase()} method orchestrates the seeding process by calling individual seeding methods for each entity.
+ * </p>
+ * <p>
+ * This component is annotated with {@code @Component} to mark it as a Spring-managed component
+ * and enable automatic detection and registration by Spring's component scanning mechanism.
+ *</p>
+ *
+ * @author Mircea Bardan
+ */
+@Component
+public class UserDatabaseSeeder {
+    private final UserRepository userRepository;
+    private final BCryptPasswordEncoder passwordEncoder;
+    private final PreferredScheduleRepository preferredScheduleRepository;
+    private final PreferredMeetingPlaceRepository preferredMeetingPlaceRepository;
+
+    private Random random = new Random(); 
+    /**
+     * Constructs a new instance of DatabaseSeeder.
+     *
+     * @param userRepository  The repository for managing user entities.
+     * @param passwordEncoder The password encoder used to encode user passwords before storing them in the database.
+     * @param preferredScheduleRepository
+     */
+    public UserDatabaseSeeder(
+        @Autowired UserRepository userRepository,
+        @Autowired BCryptPasswordEncoder passwordEncoder,
+        @Autowired PreferredScheduleRepository preferredScheduleRepository,
+        @Autowired PreferredMeetingPlaceRepository preferredMeetingPlaceRepository
+        ){
+        this.userRepository = userRepository;
+        this.passwordEncoder = passwordEncoder;
+        this.preferredScheduleRepository = preferredScheduleRepository;
+        this.preferredMeetingPlaceRepository = preferredMeetingPlaceRepository;
+    }
+
+    /**
+     * Seeds the database with initial sample data.
+     * This method delegates the seeding process to individual seeding methods for different entities.
+     */
+    public void seedDatabase(){
+        List<User> users = createUsers();
+        seedUsers(users);
+
+        seedPreferredSchedules();
+        
+        seedPreferredMeetingPlaces();
+    }
+
+    /**
+     * This method creates user entities.
+     * Each user is initialized with sample data such as email, password, role, status, and creation timestamp.
+     * @return a list of users.
+     */
+    private List<User> createUsers(){
+        User first = new User();
+        first.setEmail("mbardan@email.ro");
+        first.setAlias("Koroviev");
+        first.setPassword(passwordEncoder.encode("pass1"));
```

Comment:
To avoid this, you can put the 5 raw passwords inside the .env file, like so: 

# seed users passwords
FIRST_PASS = 'pass1'
SECOND_PASS = 'pass2'
THIRD_PASS = 'pass3'
FOURTH_PASS = 'pass4'
FIFTH_PASS = 'pass5'

then, pass these values to the application.properties:

# seed users passwords
first.pass = ${FIRST_PASS}
second.pass = ${SECOND_PASS}
third.pass = ${THIRD_PASS}
fourth.pass = ${FOURTH_PASS}
fifth.pass = ${FIFTH_PASS}

and use them in the DatabaseSeeder class, like this:

@Value("${first.pass}")
private String pass1;

@Value("${second.pass}")
private String pass2;
    
@Value("${third.pass}")
private String pass3;
    
@Value("${fourth.pass}")
private String pass4;
    
@Value("${fifth.pass}")
private String pass5;

... ... ... ... ... 

firstUser.setPassword(passwordEncoder.encode(pass1));
    
secondUser.setPassword(passwordEncoder.encode(pass2));

... ... ... ... ... etc